### PR TITLE
bug fixes for octomap

### PR DESCRIPTION
**Pull request by yantrabuddhi**

### CODE REVIEW

File: opencog/timeoctomap/TimeOctomap.cc [hunk 1]
```diff
@@ -58,22 +65,28 @@ TimeOctomap::get_time_resolution()
 bool
 TimeOctomap::get_current_time_range(time_pt& time_p, duration_c& duration)
 {
-    if (time_circle.size() < 1)
+  std::lock_guard<std::mutex> lgm(mtx);
```
Comment (linas) [inline opencog/timeoctomap/TimeOctomap.cc]: bad indentation, here. Bad whitespace, below. Stray comment in the block.

File: opencog/timeoctomap/TimeOctomap.cc [hunk 2]
```diff
@@ -99,20 +112,54 @@ TimeOctomap::step_time_unit()
...
+{
+  return auto_step;
+}
+
+void
+TimeOctomap::auto_step_time(bool astep)
+{
+  std::lock_guard<std::mutex> t_mtx(mtx_auto);
+    if (auto_step==astep)return;
+    auto_step=astep;
```
Comment (linas) [inline opencog/timeoctomap/TimeOctomap.cc]: indentation and whitespace problems.

File: opencog/timeoctomap/TimeOctomap.cc [hunk 3]
```diff
@@ -99,20 +112,54 @@ TimeOctomap::step_time_unit()
...
+  std::lock_guard<std::mutex> lgm(mtx);
     //if (!created_once)return false;
     OC_ASSERT(created_once);
     int i = time_circle.capacity() - 1;
     if (time_circle.size() < time_circle.capacity()) i = time_circle.size() - 1;
     //if (!time_circle[i].has_map(handle)) return false;//may assert too
     time_circle[i].map_tree.updateNode(location, true);
     time_circle[i].map_tree.setNodeData(location, ato);
+  //;
     return true;
```
Comment (linas) [inline opencog/timeoctomap/TimeOctomap.cc]: stray empty comment.

File: opencog/timeoctomap/TimeOctomap.cc [hunk 4]
```diff
@@ -121,10 +168,12 @@ TimeOctomap::remove_atom_at_current_time_by_location(
                                        const point3d location)
 {
     OC_ASSERT(created_once);
+  std::lock_guard<std::mutex> lgm(mtx);
     int i = time_circle.capacity() - 1;
     if (time_circle.size() < time_circle.capacity()) i = time_circle.size() - 1;
     //time_circle[i].map_tree[map_handle].setNodeData(location,UndefinedHandle);
     time_circle[i].map_tree.updateNode(location, false);
+  //;
```
Comment (linas) [inline opencog/timeoctomap/TimeOctomap.cc]: stray empty comment

File: opencog/timeoctomap/TimeOctomap.cc [hunk 5]
```diff
@@ -133,11 +182,15 @@ TimeOctomap::remove_atom_at_time_by_location(time_pt tp,
                                 const point3d location)
 {
     OC_ASSERT(created_once);
+  std::lock_guard<std::mutex> lgm(mtx);
     auto it = std::find(std::begin(time_circle), std::end(time_circle), tp); //time_circle.begin(),time_circle.end()
-    if (it == std::end(time_circle))
+    if (it == std::end(time_circle)){
```
Comment (linas) [inline opencog/timeoctomap/TimeOctomap.cc]: missing whitespace between close-paren and open-brace

File: opencog/timeoctomap/pointmemory/pointmem.scm [hunk 6]
```diff
@@ -6,8 +6,20 @@
...
+(define-public (space-nodes at-loc-link) (cog-outgoing-set (cadr (cog-outgoing-set (cadr (cog-outgoing-set at-loc-link))))))
+(define-public (loc-link-x at-loc-link) 
+		(string->number (cog-name (car (space-nodes at-loc-link))))
+)
+(define-public (loc-link-y at-loc-link)
+                (string->number (cog-name (cadr (space-nodes at-loc-link))))
+)
+(define-public (loc-link-z at-loc-link)
+                (string->number (cog-name (caddr (space-nodes at-loc-link))))
+)
```
Comment (linas) [inline opencog/timeoctomap/pointmemory/pointmem.scm]: Are these really needed? If you already have an atom that has all the information in it, why do you then need to convert it to scheme numbers?

**Comment by linas** [comment]
This looks just fine for merging, but the stray comments `//;` should be cleaned up, and the indentation should be made uniform.

Style hint: if you just have an if-statement that does only one thing, its easier to read and understand if it is written on one line: for example
```
   if (foobar != barfoo) return false;
```
instead of
```
    if (foobar != barfoo) {
       return false;
   }
```

**Comment by linas** [comment]
why did you close this? Do you no longer need this?

**Comment by linas** [comment]
As far as I can tell, this is the actual code that you want to merge, except that it still has the indentation issues that seem to be corrected in pull req #2262 ... do you want to merge that one, or this one? Or neither? Or both?

**Comment by yantrabuddhi** [comment]
fixed indentation and empty comments

**Comment by linas** [comment]
thanks

**Pull request merged**